src/System.Runtime/tests/System/String.SplitTests.cs

```diff
@@ -516,4 +496,22 @@ private static string[] ToStringArray(char[] source)
             return result;
         }
     }
+
+    // The below extension methods allow the tests to run on targets that *do not* have the new Split overloads.
+    // On targets that *do* have the new Split overloads (e.g. netcoreapp1.1), the actual overloads on String
+    // are called instead of the extension methods below (due to the way extension methods work).
+    internal static class StringSplitExtensions
```

Comment:
Do we have any way to verify that at runtime? I'm afraid that we may end up in a place where we accidentally start using these in the future and not catch it. If we don't have any clever way to do that I would prefer we put these in a separate file and condition the inclusion of that file in the project.